exp/api/v1alpha4/azuremanagedcontrolplane_webhook.go

```diff
@@ -291,3 +296,147 @@ func (r *AzureManagedControlPlane) validateSSHKey() error {
 
 	return nil
 }
+
+func (r *AzureManagedControlPlane) validateAadProfile() error {
+	if r.Spec.AADProfile != nil {
+		managedAad, err := r.validateManagedAadProfile()
+		if err != nil {
+			return err
+		}
+		legacyAad, err := r.validateLegacyAadProfile()
+		if err != nil {
+			return err
+		}
+		if managedAad && legacyAad {
+			return errors.New("conflicting values provided in AADProfile")
+		}
+	}
+	return nil
+}
+
+func (r *AzureManagedControlPlane) validateManagedAadProfile() (bool, error) {
+	if r.Spec.AADProfile.Managed != nil && *r.Spec.AADProfile.Managed && (r.Spec.AADProfile.AdminGroupObjectIDs == nil || len(*r.Spec.AADProfile.AdminGroupObjectIDs) == 0) {
+		return false, errors.New("AdminGroupObjectIDs fields missing in AADProfile")
+	}
+
+	if r.Spec.AADProfile.Managed == nil && r.Spec.AADProfile.AdminGroupObjectIDs != nil && len(*r.Spec.AADProfile.AdminGroupObjectIDs) != 0 {
+		return false, errors.New("Managed field missing in AADProfile")
+	}
+	return r.Spec.AADProfile.Managed != nil && *r.Spec.AADProfile.Managed, nil
+}
+
+func (r *AzureManagedControlPlane) validateLegacyAadProfile() (bool, error) {
+	err := ""
+	if r.Spec.AADProfile.ClientAppID != nil || r.Spec.AADProfile.ServerAppID != nil || r.Spec.AADProfile.ServerAppSecret != nil || r.Spec.AADProfile.TenantID != nil {
+		if r.Spec.AADProfile.ClientAppID == nil {
+			err = err + "missing ClientAppID in AADProfile. "
+		}
+		if r.Spec.AADProfile.ServerAppID == nil {
+			err = err + "missing ServerAppID in AADProfile. "
+		}
+		if r.Spec.AADProfile.ServerAppSecret == nil {
+			err = err + "missing ServerAppSecret in AADProfile. "
+		}
+		if r.Spec.AADProfile.TenantID == nil {
+			err = err + "missing TenantID in AADProfile. "
+		}
+	}
+	if err != "" {
+		return false, errors.New(err)
+	}
+	return r.Spec.AADProfile.ClientAppID != nil, nil
+}
+
+func (r *AzureManagedControlPlane) validateUpdateAadProfile(old *ManagedClusterAADProfile) field.ErrorList {
+	allErrs := field.ErrorList{}
+	new := r.Spec.AADProfile
+
+	if old != nil && new == nil {
+		allErrs = append(allErrs,
+			field.Invalid(
+				field.NewPath("Spec", "AADProfile"),
+				new,
+				"field cannot be nil, cannot disable AADProfile"))
+	}
+
+	if old == nil && new != nil {
```

Comment:
does this section mean for legacy you can't enable aad after cluster is created?

Reply:
yes correct